test/pipelinerun_test.go

```diff
@@ -192,6 +192,25 @@ func TestPipelineRun(t *testing.T) {
 			if len(events) != td.expectedNumberOfEvents {
 				t.Fatalf("Expected %d number of successful events from pipelinerun and taskrun but got %d", td.expectedNumberOfEvents, len(events))
 			}
+
+			logger.Info("Making sure the expected TaskRuns Status are included in PipelineRun Status")
+			run, err := c.PipelineRunClient.Get(prName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Can't get PipelineRun %s becuase of: %s", prName, err)
+			}
+			if len(run.Status.TaskRuns) != len(td.expectedTaskRuns) {
+				t.Fatalf("PipelineRun Status does not include TaskRuns: want: %d got %d", len(td.expectedTaskRuns), len(run.Status.TaskRuns))
+			}
+			for _, taskRunsName := range expectedTaskRunNames {
+				if len(run.Status.TaskRuns) == 0 {
+					t.Fatalf("Expected taskrun status for %s in PipelineRun %s", taskRunsName, prName)
+				}
+				status := run.Status.TaskRuns[taskRunsName]
+				if !status.GetCondition(duckv1alpha1.ConditionSucceeded).IsTrue() {
+					t.Fatalf("Expected TaskRun %s to have succeeded but Status is %v", taskRunsName, status)
+				}
```

Comment:
What do you think about moving this test down a layer or two, either testing it at the controller level or in a more focused unit test?

_or_ leave this here but add some extra coverage via unit tests? (if you do keep this here, i think it might be more readable if it was in a separate function? the test is a bit long)

Reply:
I'd like to keep it here since this function is in the reconciler so the unit test might not be the best representation of it, but I will reduce some of the checks

Reply:
i want to push back on this again - i think we're really testing way more than we need to in our integration tests and would very much like to see this coverage go down to the unit tests instead - the more test at the level of the end to end and reconciler tests, the harder it is going to be to maintain these tests and change them over time

i would settle for a compromise of adding coverage at the unit test level also but eventually you'll probably see a PR from me where i try to delete these checks here ❤️❤️❤️

Reply:
added unit test and removed the checks in the integration test